sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -417,6 +417,14 @@ object NullPropagation extends Rule[LogicalPlan] {
         case left :: Literal(null, _) :: Nil => Literal.create(null, e.dataType)
         case _ => e
       }
+
+      // If the value expression is NULL then transform the In expression to
+      // Literal(null)
+      case e @ In(v, list) => v match {
```

Comment:
@cloud-fan -
If i re-order the rules to move ConstantFolding ahead of NullPropagation then it works fine.
I don't know the impact of doing this .. need your advice.